scripts/asset_processor/video_asset_processor.py

```diff
@@ -23,630 +23,570 @@
-				elif label.split('-')[0] in upscale_features:
-					df_features[label] = df_features[label] * df_features['dimension']
-					logger.debug(f'Upscaling {label}')
-		return df_features
-
-	def process(self):
-		"""
-		Function to aggregate computed values of metrics
-		of iterated renditions into a pandas DataFrame.
-		"""
-		if self.do_process:
-			if self.do_profiling:
-				self.capture_to_array = self.cpu_profiler(self.capture_to_array)
-				self.compare_renditions_instant = self.cpu_profiler(self.compare_renditions_instant)
-			# Iterate through renditions
-			for rendition in self.renditions_list:
-				path = rendition['path']
-				capture = None
-				try:
-					if os.path.exists(path):
-						capture = VideoCapture(path, use_gpu=self.use_gpu)
-						# Turn openCV capture to a list of numpy arrays
-						master_idx_map, frame_list, frame_list_hd, pixels, height, width = self.capture_to_array(capture)
-						dimensions = '{}:{}'.format(int(width), int(height))
-						# Compute the metrics for the rendition
-						self.metrics[path] = self.compute(master_idx_map,
-														  frame_list,
-														  frame_list_hd,
-														  path,
-														  dimensions,
-														  pixels)
-					else:
-						logger.error(f'Unable to find rendition file: {path}')
-				except Exception as err:
-					logger.exception('Unable to compute metrics for {}'.format(path))
-				finally:
-					if capture is not None:
-						capture.release()
-
-			if self.do_profiling:
-				self.cpu_profiler.print_stats()
-			if self.debug_frames:
-				print(f'Frames metrics of {type(self).__name__}')
-				print(self._convert_debug_metrics(self.metrics))
-			return self.aggregate(self.metrics)
-		else:
-			logger.error('Unable to process. Original source path does not exist')
-			return False
-
-	@staticmethod
-	def _convert_debug_metrics(metrics):
-		res = []
-		for f, frames in metrics.items():
-			for id, frame in frames.items():
-				row = {'file': f, 'sample': id}
-				row.update(frame)
-				res.append(row)
-		df = pd.DataFrame(res)
-		df.set_index(['file', 'sample'], inplace=True)
-		df.sort_index(inplace=True)
-		return df
-
-	def read_renditions_metadata(self):
-		# Iterate through renditions
-		last_rendition_fps = None
-		for rendition in self.renditions_list:
-			path = rendition['path']
-			try:
-				if os.path.exists(path):
-					capture = VideoCapture(path, use_gpu=False)
-					# Get framerate
-					fps = capture.fps
-					# Validate frame rates, only renditions with same FPS (though not necessarily equal to source video) are currently supported in a single instance
-					if last_rendition_fps is not None and last_rendition_fps != fps:
-						raise Exception(f'Rendition has frame rate incompatible with other renditions: {fps}')
-					rendition['fps'] = fps
-					rendition['width'] = capture.width
-					rendition['height'] = capture.height
-				else:
-					raise Exception(f'Rendition not found: {path}')
-			finally:
-				capture.release()
+    """
```

Comment:
Could we match the indentation of the file and then if indentation consistency is a problem in the repo we could update the indentation to be consistent across all files in a separate PR? This would make it easier to distinguish functional vs. formatting changes for review.

Reply:
Indentation is consistent now, I believe. Would be great to also follow PEP8, and maybe add a test to check compliance. Actually, it's just that particular file I have converted to class, and then reformatted properly in a separate commit.